pkg/proxy/conntrack/conntrack.go

```diff
@@ -49,95 +50,139 @@ type Interface interface {
-	}
-	klog.V(4).InfoS("Conntrack entries deleted", "output", string(output))
-	return nil
+	return 0
 }
 
 // ClearEntriesForIP is part of Interface
-func (ct *execCT) ClearEntriesForIP(ip string, protocol v1.Protocol) error {
-	parameters := parametersWithFamily(utilnet.IsIPv6String(ip), "-D", "--orig-dst", ip, "-p", protoStr(protocol))
-	err := ct.exec(parameters...)
-	if err != nil && !strings.Contains(err.Error(), noConnectionToDelete) {
+func (ct *conntracker) ClearEntriesForIP(ip string, protocol v1.Protocol) error {
+	filter := &conntrackFilter{
+		protocol: getProtocolNumber(protocol),
+		original: &connectionTuple{
+			dstIP: netutils.ParseIPSloppy(ip),
+		},
+	}
+	klog.V(4).InfoS("Clearing conntrack entries", "ip", ip, "protocol", protocol)
+
+	n, err := ct.handler.ConntrackDeleteFilters(netlink.ConntrackTable, getNetlinkFamily(netutils.IsIPv6String(ip)), filter)
+	if err != nil {
 		// TODO: Better handling for deletion failure. When failure occur, stale udp connection may not get flushed.
 		// These stale udp connection will keep black hole traffic. Making this a best effort operation for now, since it
 		// is expensive to baby-sit all udp connections to kubernetes services.
-		return fmt.Errorf("error deleting connection tracking state for UDP service IP: %s, error: %v", ip, err)
+		return fmt.Errorf("error deleting connection tracking state for %s service IP: %s, error: %w", protoStr(protocol), ip, err)
 	}
+	klog.V(4).InfoS("Cleared conntrack entries", "count", n)
 	return nil
 }
 
-// ClearEntriesForPort is part of Interface
-func (ct *execCT) ClearEntriesForPort(port int, isIPv6 bool, protocol v1.Protocol) error {
+// ClearEntriesForPort delete the conntrack entries for connections specified by the port.
+func (ct *conntracker) ClearEntriesForPort(port int, isIPv6 bool, protocol v1.Protocol) error {
+	filter := &conntrackFilter{
+		protocol: getProtocolNumber(protocol),
+		original: &connectionTuple{
+			dstPort: uint16(port),
+		},
+	}
 	if port <= 0 {
 		return fmt.Errorf("wrong port number. The port number must be greater than zero")
 	}
-	parameters := parametersWithFamily(isIPv6, "-D", "-p", protoStr(protocol), "--dport", strconv.Itoa(port))
-	err := ct.exec(parameters...)
-	if err != nil && !strings.Contains(err.Error(), noConnectionToDelete) {
-		return fmt.Errorf("error deleting conntrack entries for UDP port: %d, error: %v", port, err)
+
+	klog.V(4).InfoS("Clearing conntrack entries", "port", port, "protocol", protocol)
+	n, err := ct.handler.ConntrackDeleteFilters(netlink.ConntrackTable, getNetlinkFamily(isIPv6), filter)
+	if err != nil {
+		return fmt.Errorf("error deleting connection tracking state for %s port: %d, error: %w", protoStr(protocol), port, err)
+	}
+	if n == 0 {
+		return fmt.Errorf("error deleting connection tracking state for %s port: %d, no entries found", protoStr(protocol), port)
 	}
 	return nil
 }
 
 // ClearEntriesForNAT is part of Interface
```

Comment:
```suggestion
// ClearEntriesForNAT delete the conntrack entries for connections which had been DNATted from origin to dest
```

(I just copied the Interface comments here)